src/server/req_quejob.c

```diff
@@ -1003,7 +1003,7 @@ req_quejob(struct batch_request *preq)
 	 * job structure and attributes already set up.
 	 */
 
-	rc = svr_chkque(pj, pque, preq->rq_host, MOVE_TYPE_Move);
+	rc = svr_chkque(pj, pque, pj->ji_wattr[(int)JOB_ATR_submit_host].at_val.at_str, MOVE_TYPE_Move);
```

Comment:
Is it possible for this attribute to be unset?  Do you need to check if it is set first and then use preq if it isn't?

Reply:
It is set on job creation and it never changes. It is similar to JOB_ATR_ctime. The source for this attribute is conn->cn_physhost, which comes from request->rq_host (before changing to the realm). IMHO, it is safe to consider the attribute is always set.